src/core/base.jl

```diff
@@ -1,85 +1,50 @@
-# stuff that is universal to all gas grid models
-
-export
-    setdata, solve,
-    run_generic_model, build_generic_model, solve_generic_model
-
-" Do a solve of the problem "
-function JuMP.solve(gm::GenericGasModel)
-    status, solve_time, solve_bytes_alloc, sec_in_gc = @timed solve(gm.model)
-    try
-        solve_time = getsolvetime(gm.model)
-    catch
-        @warn "there was an issue with getsolvetime() on the solver, falling back on @timed.  This is not a rigorous timing value."
-    end
-
-    return status, solve_time
-end
-
-""
-function run_generic_model(power_file, gas_file, power_model_constructor, gas_model_constructor, solver, post_method; solution_builder = get_solution, kwargs...)
-    power_data    = PowerModels.parse_file(power_file)
-    gas_data      = GasModels.parse_file(gas_file)
-    return run_generic_model(power_data, gas_data, power_model_constructor, gas_model_constructor, solver, post_method; solution_builder = solution_builder, kwargs...)
-end
-
-" Run the optimization on a dictionarized model"
-function run_generic_model(power_data::Dict{String,Any}, gas_data::Dict{String,Any}, power_model_constructor, gas_model_constructor, solver, post_method; power_ref_extensions=[], solution_builder = get_solution, kwargs...)
-    pm, gm = build_generic_model(power_data, gas_data, power_model_constructor, gas_model_constructor, post_method; power_ref_extensions=power_ref_extensions, kwargs...)
-    solution = solve_generic_model(pm, gm, solver; solution_builder = solution_builder)
-    return solution
-end
-
 ""
-function build_generic_model(pfile::String, gfile::String, power_model_constructor, gas_model_constructor, post_method; kwargs...)
-    gas_data = GasModels.parse_file(gfile)
-    power_data = PowerModels.parse_file(pfile)
-    return build_generic_model(power_data, gas_data, power_model_constructor, gas_model_constructor, post_method; kwargs...)
-end
-
-" Dummy post method to do nothing for the individual system"
-function empty_post_method(m; kwargs...)
+function instantiate_model(gfile::String, pfile::String, gtype::Type, ptype::Type, build_method; gext=[], pext=[], kwargs...)
+    gdata, pdata = [_GM.parse_file(gfile), _PM.parse_file(pfile)]
```

Comment:
nit, this syntax is a little odd.  You allocate an array for two objects and then immediately release an unallocate it.

I would recomend,
```
gdata = _GM.parse_file(gfile)
pdata = _PM.parse_file(pfile)
```